### PR TITLE
[terra-slide-panel] Turn focus outline off; replace transition logic

### PR DESCRIPTION
**Pull request by yuderekyu**
### Summary
Closes https://github.com/cerner/terra-framework/issues/609, https://github.com/cerner/terra-framework/issues/606, https://github.com/cerner/terra-framework/issues/447

#### Changes
* Set main + content panel outline to none
* Move focus transition logic to `componentDidUpdate`

### CODE REVIEW

File: packages/terra-slide-panel/src/SlidePanel.jsx [hunk 1]
```diff
@@ -58,53 +58,24 @@ class SlidePanel extends React.Component {
     super(props);
     this.setPanelNode = this.setPanelNode.bind(this);
     this.mainNode = React.createRef();
-    this.handleTransitionEnd = this.handleTransitionEnd.bind(this);
-    this.preparePanelForTransition = this.preparePanelForTransition.bind(this);
 
     this.isHidden = !props.isOpen;
```
Comment (tbiethman) [inline packages/terra-slide-panel/src/SlidePanel.jsx]: From what I'm seeing, I don't think we need to manage this internal property anymore?

Comment (tbiethman) [reply packages/terra-slide-panel/src/SlidePanel.jsx]: without animations, isHidden is just !isOpen all the time.

Comment (yuderekyu) [reply packages/terra-slide-panel/src/SlidePanel.jsx]: https://github.com/cerner/terra-framework/pull/617/commits/f8bc8503300c1f9a66e7a18aa33b80ed7cf64d3b

Comment (yuderekyu) [reply packages/terra-slide-panel/src/SlidePanel.jsx]: wait a sec....do we even need to toggle `aria-hidden` inside componentUpdate if we use `isOpen` inside the render function?

Comment (tbiethman) [reply packages/terra-slide-panel/src/SlidePanel.jsx]: I don't think so. Remove all the things.

**Pull request merged**